modules/dbparser/tests/test_patternize.c

```diff
@@ -31,18 +31,23 @@
 #include <stdio.h>
 #include <string.h>
 
-gboolean fail = FALSE;
+#include <criterion/criterion.h>
+#include <criterion/parameterized.h>
 
 MsgFormatOptions parse_options;
 
-static void _debug_print(gpointer key, gpointer value, gpointer dummy)
+static void setup(void)
 {
-  fprintf(stderr, "%s: %d\n", (gchar *) key, *((guint *) value));
+  app_startup();
+  configuration = cfg_new_snippet();
+  cfg_load_module(configuration, "syslogformat");
+  msg_format_options_defaults(&parse_options);
+  msg_format_options_init(&parse_options, configuration);
 }
 
-static void _debug_print2(gpointer key, gpointer value, gpointer dummy)
+static void teardown(void)
 {
-  fprintf(stderr, "%s: %d\n", (gchar *) key, ((Cluster *) value)->loglines->len);
+  log_tags_global_deinit();
```

Comment:
Missing deinit calls like `app_shutdown`. (The original test was also missing this.)

Reply:
Missing deinit functions added